src/SLCore.UnitTests/Common/Helpers/LanguageExtensionsTests.cs

```diff
@@ -20,47 +20,42 @@
 
 using SonarLint.VisualStudio.SLCore.Common.Helpers;
 using SonarLint.VisualStudio.SLCore.Common.Models;
+using static SonarLint.VisualStudio.Core.Language;
 
 namespace SonarLint.VisualStudio.SLCore.UnitTests.Common.Helpers;
 
 [TestClass]
 public class LanguageExtensionsTests
 {
     [TestMethod]
-    public void VerifyConversionToCoreLanguage()
-    {
-        VerifyConversionToCoreLanguage(Language.C, VisualStudio.Core.Language.C);
-        VerifyConversionToCoreLanguage(Language.CPP, VisualStudio.Core.Language.Cpp);
-        VerifyConversionToCoreLanguage(Language.CS, VisualStudio.Core.Language.CSharp);
-        VerifyConversionToCoreLanguage(Language.CSS, VisualStudio.Core.Language.Css);
-        VerifyConversionToCoreLanguage(Language.JS, VisualStudio.Core.Language.Js);
-        VerifyConversionToCoreLanguage(Language.SECRETS, VisualStudio.Core.Language.Secrets);
-        VerifyConversionToCoreLanguage(Language.TS, VisualStudio.Core.Language.Ts);
-        VerifyConversionToCoreLanguage(Language.VBNET, VisualStudio.Core.Language.VBNET);
-        VerifyConversionToCoreLanguage(Language.TSQL, VisualStudio.Core.Language.TSql);
-        VerifyConversionToCoreLanguage(Language.ABAP, VisualStudio.Core.Language.Unknown);
-        VerifyConversionToCoreLanguage(Language.JAVA, VisualStudio.Core.Language.Unknown);
-    }
+    [DynamicData(nameof(CoreToSlCore))]
+    public void ConvertToCoreLanguage_KnownLanguage_ConvertsAsExpected(VisualStudio.Core.Language coreLanguage, Language slCoreLanguage) =>
```

Comment:
add a smoke test that asserts that every language from LannguageProvider.AllKnownLanguages converts to non-unknown slcore language